src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java

```diff
@@ -43,17 +46,44 @@ public void tapOnElement(WebElement element) {
     jsexec.executeScript("arguments[0].click();", element);
   }
 
-  public void waitMilliseconds(int time, String methodName) {
+  public void waitMilliseconds(int time, String message) {
     try {
       Thread.sleep(time);
     } catch (InterruptedException e) {
-      PageObjectLogging.log(methodName, e.getMessage(), false);
+      PageObjectLogging.log(message, e.getMessage(), false);
+    }
+  }
+
+  /**
+   * First waits for spinner to be visible and then waits for spinner to be hidden
+   */
+  public void waitForLoadingSpinnerToFinishReloadingPage() {
+    try {
+      waitForElementByElement(loadingSpinner);
+      waitForElementPresenceByBy(By.cssSelector(".loading-overlay.hidden"));
+    } catch (TimeoutException e) {
+    }
+  }
+
+  /**
+   * Example: wait 10 sec for element and check each 0.5 for that element (10 / 0.5 = 20 attempts)
+   *
+   * @param element            WebElement
+   * @param timeOutInSec       int
+   * @param checkOutInMilliSec int
+   */
+  public void waitForElementVisibleByElementCustomTimeOut(WebElement element, int timeOutInSec,
```

Comment:
I need that in situation, when i had element and then it was deleted from DOM, so i want to check that this element is no longer present